### PR TITLE
feat: upgradability

### LINKED ISSUE #43: Upgraddable Function
Write a function using open zeppelin that allows  us to upgrade contract base on a new class hash and also write unit test for it

### PR DESCRIPTION
**Pull request by Akshola00**
implement and test upgrade using oz
closes #43

### CODE REVIEW

**Comment by Akshola00** [comment]
@mubarak23 pls review

File: tests/test_campaign.cairo [hunk 1]
```diff
@@ -62,6 +62,24 @@ fn __deploy_token_giver_NFT__() -> ContractAddress {
     return (nft_contract_address);
 }
 
+
+fn test_upgradability() {
```
Comment (mubarak23) [inline tests/test_campaign.cairo:66]: move this test to last section of the test file

File: tests/test_campaign.cairo [hunk 2]
```diff
@@ -62,6 +62,24 @@ fn __deploy_token_giver_NFT__() -> ContractAddress {
...
+    strk_address.serialize(ref calldata);
+
+    let (contract_address, _) = class_hash.deploy(@calldata).unwrap();
+
+    let campaign_dispatcher = ICampaignDispatcher { contract_address };
+    let new_class_hash = declare("TokengiverCampaign").unwrap().contract_class().class_hash;
+    campaign_dispatcher.upgrade(*new_class_hash);
+}
+
+
```
Comment (mubarak23) [inline tests/test_campaign.cairo:82]: add a test case that panic if wrong owner try to upgrade the contract

**Review by mubarak23 — CHANGES_REQUESTED** [review]

**Comment by Akshola00** [comment]
@mubarak23 pls review

**Comment by Akshola00** [comment]
@mubarak23 pushed the requested test

**Pull request merged**